Source/Extensions/UIViewExtensions.swift

```diff
@@ -76,3 +76,17 @@ extension UIView {
     }
 
 }
+
+extension UIView {
+    func constraint(withIdentifier:String) -> NSLayoutConstraint? {
+        return self.constraints.filter{ $0.identifier == withIdentifier }.first
+    }
+}
+
+extension NSLayoutAnchor {
+    @objc func constraintEqualToAnchor(anchor: NSLayoutAnchor!, constant:CGFloat, identifier:String) -> NSLayoutConstraint {
```

Comment:
To also give an alternative, you're currently trying to change a constraint identified by a given OS generated name to your requirement. However, that is a very unsafe way to do so, as it can break from:
- os identifier formatting differences
- code changes that are not linked in any way with the String identifier

Generally, if you want to modify a constraint, there are a few options
```swift
// Let's say we have this view that is part of some complex hierarchy we don't need to know of
let signatureView = UIView()

// If we want to modify the padding of this inside its parent, we can do a few things

// 1. Hold reference to constraint at creation of layout
let modifiableConstraint = NSLayoutConstraint( .... )
NSLayoutConstraint.activate([modifiableConstraint])

// 2. Navigate constraints of the view
let constraintIWant = signatureView.constraints.first(where: {
    $0.firstAttribute == .bottom &&
    ($0.firstItem == signatureView && $0.secondItem == superview ||
    $0.firstItem == superview && $0.secondItem == signatureView) 
})
```

Reply:
Let me know if I can give another example to the specific problem you're trying to solve, if this is not sufficient.

Reply:
As we discussed on Zoom you have explained to me the issue this code was fixing. I think this [SO answer](https://stackoverflow.com/a/37534822) will be a cleaner, no code solution to the issue as well.

The constraint kept in memory has to have a **strong** reference for it to not be deallocated when deactivated. As the constraint has no further reference of its own, that should be a safe behaviour that would not appear to lead to retain cycles.